packages/zql/src/query/query-impl.query.test.ts

```diff
@@ -1199,6 +1199,85 @@ test('where exists', () => {
   expect(materialized.data).toEqual([]);
 });
 
+test('duplicative where exists', () => {
+  const queryDelegate = new QueryDelegateImpl();
+  const issueSource = must(queryDelegate.getSource('issue'));
+  const labelSource = must(queryDelegate.getSource('label'));
+  const issueLabelSource = must(queryDelegate.getSource('issueLabel'));
+  issueSource.push({
+    type: 'add',
+    row: {
+      id: '0001',
+      title: 'issue 1',
+      description: 'description 1',
+      closed: false,
+      ownerId: '0001',
+    },
+  });
+  issueSource.push({
+    type: 'add',
+    row: {
+      id: '0002',
+      title: 'issue 2',
+      description: 'description 2',
+      closed: true,
+      ownerId: '0002',
+    },
+  });
+  labelSource.push({
+    type: 'add',
+    row: {
+      id: '0001',
+      name: 'bug',
+    },
+  });
+
+  const materialized = newQuery(queryDelegate, schema, 'issue')
+    .where('closed', true)
+    .whereExists('labels', q => q.where('name', 'bug'))
+    .whereExists('labels', q => q.where('name', 'bug'))
```

Comment:
huh. Don't these end up with the same relationship name? I'm surprised there are not other bugs with this.

Reply:
This logic in builder assures they end up with unique aliases, https://github.com/rocicorp/mono/blob/77b2edf0f3e69a7b16483c783f37e813c656ee94/packages/zql/src/builder/builder.ts#L389

Though, I'm not sure what happens if you duplicate related...